Updates response to display public directory.

saralein opened the pull request.

saralein committed:
Updates response to display public directory.

zdavy commented on src/main/java/com/saralein/server/response/DirectoryResponse.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    private String createBody() {
+        FileHelper fileHelper = new FileHelper("public");
```
Something about the file helper being initialized on every call to `createBody` seems unnecessary to me, and then you're tied to this specific implementation of FileHelper. It's kind of rigid.

damonkelley commented on src/main/java/com/saralein/server/response/FileHelper.java:
```diff
@@ -0,0 +1,39 @@
+    private File[] listDirectoryFiles() {
+        return getDirectoryContents().listFiles(
+            file -> !file.getName().equals(".DS_Store"));
+    }
```
It might be easier if you returned a `List<File>` here. You can easily convert any array into a List by using `Arrays.asList`.

I tend to try to avoid using arrays directory, but that is just me 🤷‍♂️

saralein replied:
[Adjusted this](https://github.com/saralein/java-server/pull/2/commits/dba3c8c50f36e057acc529d833797c6300d578c1#diff-133e41c6e2f48487e5d6277fe3ce5cf1R24).  So nice that nothing else had to be adjusted for that change to work.  Also, I think I had an experience elsewhere with array vs. list where I can see why you'd prefer lists.  Will keep it in mind moving forward.

saralein committed:
Creates interface for file helpers and injects into directory response.

zdavy commented on src/main/java/com/saralein/server/response/DirectoryResponse.java:
```diff
@@ -16,7 +21,7 @@ private String createHeader() {
 
     private String createBody() {
-        FileHelper fileHelper = new FileHelper("public");
+//        SysFileHelper fileHelper = new SysFileHelper("public");
```
I'm not sure there's much value in this comment

saralein replied:
Definitely meant to remove that.  Will do so.  Thank you.

saralein committed:
Removes extraneous comment.

saralein commented:
Alright, changed the file helper to an interface and updated what was previously the file helper to be a system file helper implementation.  Also injected it into the directory response.
